kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/HorizontalPodAutoscalerTest.java

```diff
@@ -0,0 +1,184 @@
+    assertFalse(deleted);
+
+    deleted = client.autoscaling().horizontalPodAutoscalers().inNamespace("ns1").withName("horizontalpodautoscaler2").delete();
+    assertTrue(deleted);
+  }
+
+  @Test
+  public void testDeleteMulti() {
+    HorizontalPodAutoscaler horizontalPodAutoscaler1 = new HorizontalPodAutoscalerBuilder().withNewMetadata().withName("horizontalpodautoscaler1").withNamespace("test").endMetadata().build();
+    HorizontalPodAutoscaler horizontalPodAutoscaler2 = new HorizontalPodAutoscalerBuilder().withNewMetadata().withName("horizontalpodautoscaler2").withNamespace("ns1").endMetadata().build();
+    HorizontalPodAutoscaler horizontalPodAutoscaler3 = new HorizontalPodAutoscalerBuilder().withNewMetadata().withName("horizontalpodautoscaler3").withNamespace("any").endMetadata().build();
+
+    server.expect().withPath("/apis/autoscaling/v2beta1/namespaces/test/horizontalpodautoscalers/horizontalpodautoscaler1").andReturn(200, horizontalPodAutoscaler1).once();
+    server.expect().withPath("/apis/autoscaling/v2beta1/namespaces/ns1/horizontalpodautoscalers/horizontalpodautoscaler2").andReturn(200, horizontalPodAutoscaler2).once();
+
+    KubernetesClient client = server.getClient();
+    Boolean deleted = client.autoscaling().horizontalPodAutoscalers().inAnyNamespace().delete(horizontalPodAutoscaler1, horizontalPodAutoscaler2);
+    assertTrue(deleted);
+
+    deleted = client.autoscaling().horizontalPodAutoscalers().inAnyNamespace().delete(horizontalPodAutoscaler3);
+    assertFalse(deleted);
+  }
+
+  @Test(expected = KubernetesClientException.class)
+  public void testCreateWithNameMismatch() {
+    HorizontalPodAutoscaler horizontalPodAutoscaler1 = new HorizontalPodAutoscalerBuilder().withNewMetadata().withName("horizontalpodautoscaler1").withNamespace("test").endMetadata().build();
+    KubernetesClient client = server.getClient();
+    client.autoscaling().horizontalPodAutoscalers().inNamespace("test").withName("horizontalpodautoscaler1").create(horizontalPodAutoscaler1);
+  }
+
+  @Test
+  public void testLoadFromFile() {
+    KubernetesClient client = server.getClient();
+    HorizontalPodAutoscaler horizontalPodAutoscaler = client.autoscaling().horizontalPodAutoscalers().load(getClass().getResourceAsStream("/test-horizontalpodautoscaler.yml")).get();
+    assertEquals("php-apache", horizontalPodAutoscaler.getMetadata().getName());
+  }
+
+  @Test
+  public void testBuild() {
+    HorizontalPodAutoscaler horizontalPodAutoscaler = new HorizontalPodAutoscalerBuilder()
+      .withNewMetadata().withName("test-hpa").withNamespace("test").endMetadata()
+      .withNewSpec()
+      .withNewScaleTargetRef()
+      .withApiVersion("apps/v1")
+      .withKind("Deployment")
+      .withName("the-deployment")
+      .endScaleTargetRef()
+      .withMinReplicas(1)
+      .withMaxReplicas(10)
+      .addToMetrics(new io.fabric8.kubernetes.api.model.MetricSpecBuilder()
```

Comment:
You need to modify it since #1590 has been merged and HPA's spec has changed a bit. Rebase this branch with master and rebuild, you would be able to see errors.

Reply:
@rohanKanojia: Requested changes made.